go/obscuronode/host/host.go

```diff
@@ -460,11 +466,18 @@ func (a *Node) requestSecret() {
 			txs := b.b.DecodeBlock().Transactions()
 			for _, tx := range txs {
 				t := a.txHandler.UnPackTx(tx)
-				if t != nil && t.TxType == obscurocommon.StoreSecretTx && t.Attestation.Owner == a.ID {
-					// someone has replied
-					log.Log(fmt.Sprintf("Node-%d: Secret was retrieved", obscurocommon.ShortAddress(a.ID)))
-					a.Enclave.InitEnclave(t.Secret)
-					return
+				if t != nil && t.TxType == obscurocommon.StoreSecretTx {
+					att, err := nodecommon.DecodeAttestation(t.Attestation)
+					if err != nil {
+						log.Log("Failed to decode attestation report")
+						return
```

Comment:
Does this need the error message printed on ?

Reply:
Yeah good shout. It's waiting for a secret and whatever someone sent didn't work, be useful to know why...